testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -481,12 +630,14 @@ impl RelayerThread {
             .expect("FATAL: unknown consensus hash");
 
         // always clear this even if this isn't the latest sortition
-        let won_sortition = sn.sortition && self.last_commits.remove(&sn.winning_block_txid);
+        self.last_commits.remove(&sn.winning_block_txid);
+        let won_sortition = sn.sortition; // && cleared;
```

Comment:
This does not seem right. Is this intended to be this instead?

```rust
        let cleared = self.last_commits.remove(&sn.winning_block_txid);
        let won_sortition = sn.sortition && cleared;
```

Reply:
`miner_forking` passes after I make this change.

Reply:
`tenure_extend_after_bad_commit` and `continue_after_fast_block_no_sortition` also pass with this change. 

`tenure_extend_after_failed_miner` is the only remaining failure.

Reply:
Fixed in 9de3f8412.